internal/common/ldproc_test.go

```diff
@@ -0,0 +1,141 @@
+package common
+
+import (
+	"bytes"
+
+	"encoding/json"
+	"fmt"
+	"github.com/piprate/json-gold/ld"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// expectations is in identifiers_test
+
+/*
+this tests a single path against a single json file
+*/
+func TestNormalizeTriple(t *testing.T) {
+	var jsonNoContext = `{
+"@type":"bar",
+"@id":"idenfitier",
+"url": "http://example.com/",
+"identifier": [	
+	{
+	"@type": "PropertyValue",
+	"@id": "https://doi.org/10.1575/1912/bco-dmo.2343.1",
+	"propertyID": "https://registry.identifiers.org/registry/doi",
+	"value": "doi:10.1575/1912/bco-dmo.2343.1",
+	"url": "https://doi.org/10.1575/1912/bco-dmo.2343.1"
+	}
+	
+]
+
+}`
+	var jsonNoContextSimple = `
+        {
+            "@type":"bar",
+            "SO:name":"Some type in a graph"
+        }
+`
+	var jsonGraphFirst = `{
+    "@graph":[
+        {
+            "@context": {
+                "SO":"http://schema.org/"
+            },
+            "@type":"bar",
+            "SO:name":"Some type in a graph"
+        }
+    ]
+}`
+	var tests = []expectations{
+		// default
+		{
+			name:          "noContext",
+			json:          map[string]string{"jsonID": jsonNoContext},
+			errorExpected: false,
+
+			expected: "[idenfitier]",
+			ignore:   false,
+		},
+		{
+			name:          "noContextSimple",
+			json:          map[string]string{"jsonID": jsonNoContextSimple},
+			errorExpected: false,
+
+			expected: "_:c14n0 <SO:name> \"Some type in a graph\" .\n_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <bar> .\n",
+			ignore:   false,
+		},
+		{
+			name:          "jsonGraphFirst",
+			json:          map[string]string{"jsonID": jsonGraphFirst},
+			errorExpected: false,
+
+			expected: "_:c14n0 <http://schema.org/name> \"Some type in a graph\" .\n_:c14n0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <bar> .\n",
+			ignore:   false,
+		},
+	}
+	testNormalizeTriple(tests, t)
+}
+
+func testNormalizeTriple(tests []expectations, t *testing.T) {
+	var vipercontext = []byte(`
+context:
```

Comment:
Do you need all this stuff in the context here? It would help me understand what the test is doing better if the context only had what was necessary to run the test.

Reply:
Fixed a bit. Trimmed context, and first test should have been empty. 
Tests used to determine when NormalizeTriples returns an empty document.
 
(Future) Need deeper testing or proxying/wrapping NormalizeTriples of this since empty or same set of triples being returned from normalize causes  identifier conflict.  
Identifier addresses the empty case and hashes the original json to create a more unique id.